app/src/main/java/org/dhis2/Bindings/DataElementValuesExtensions.kt

```diff
@@ -0,0 +1,31 @@
+package org.dhis2.Bindings
+
+import android.graphics.Color
+import android.text.SpannableString
+import android.text.SpannableStringBuilder
+import android.text.Spanned
+import android.text.style.ForegroundColorSpan
+
+fun List<Pair<String, String>>.toSpannableString(): SpannableStringBuilder {
+    val stringBuilder = SpannableStringBuilder()
+    this.forEach { nameValuePair ->
+        if (nameValuePair.second != "-") {
+            val value = SpannableString(nameValuePair.second)
+            val colorToUse = when {
+                this.indexOf(nameValuePair) % 2 == 0 -> Color.parseColor("#8A333333")
+                else -> Color.parseColor("#61333333")
+            }
+            value.setSpan(
+                ForegroundColorSpan(colorToUse),
+                0,
+                value.length,
+                Spanned.SPAN_INCLUSIVE_INCLUSIVE
+            )
```

Comment:
why an extension? I think the presentation logic could be decouple from android dependencies so you can unit test.
Let's talk about it in the daiy :)